### PR TITLE
Fix non-revealed chapters in ModuleView

### LINKED ISSUE #1266: Course chapters not accessible in module view
In the course module view the sub-chapters are shown as not accessible even though they should, [see for example here in A+ manual](https://plus.cs.aalto.fi/aplus-manual/master/set_up_environment/). This seems to relate to the [recent modification](https://github.com/apluslms/a-plus/commit/ae9258302f299a3cf148557ee10478c5c4c877ce) that applies revel rules to course chapters, and uses the new is_revealed attribute when deciding on chapter accessibility.

This is particularly harmful in situations when a course module is linked as LTI activity (e.g. from MyCourses), because the chapters in the module have become unaccessible.

### PR DESCRIPTION
**Pull request by PasiSa**
Fixes #1266

# Description

**What?**

[ANSWER HERE]

**Why?**

[ANSWER HERE]

**How?**

[ANSWER HERE]

Fixes #<issue number>

# Testing

**Remember to add or update unit tests for new features and changes.**

* How to [test your changes in A-plus](https://github.com/apluslms/a-plus/tree/master/doc#running-tests-and-updating-translations)
* How to [test accessibility](https://wiki.aalto.fi/display/EDIT/How+to+check+the+accessibility+of+pull+requests)

**What type of test did you run?**

- [ ] Accessibility test using the [WAVE](https://wave.webaim.org/extension/) extension.
- [ ] Django unit tests.
- [ ] Selenium tests.
- [ ] Other test. *(Add a description below)*
- [ ] Manual testing.

[ADD A DESCRIPTION ABOUT WHAT YOU TESTED MANUALLY]

**Did you test the changes in**

- [ ] Chrome
- [ ] Firefox
- [ ] This pull request cannot be tested in the browser.

**Think of what is affected by these changes and could become broken**

# Translation

- [ ] Did you modify or add new strings in the user interface? ([Read about how to create translation](https://github.com/apluslms/a-plus/tree/master/doc#running-tests-and-updating-translations))

# Programming style

- [ ] Did you follow our [style guides](https://apluslms.github.io/contribute/styleguides/)?
- [ ] Did you use Python type hinting in all functions that you added or edited? ([type hints](https://docs.python.org/3/library/typing.html) for function parameters and return values)

# Have you updated the README or other relevant documentation?

- [ ] documents inside the doc directory.
- [ ] README.md.
- [ ] Aplus Manual.
- [ ] Other documentation (mention below which documentation).

# Is it Done?

- [ ] Reviewer has finished the code review
- [ ] After the review, the developer has made changes accordingly
- [ ] Customer/Teacher has accepted the implementation of the feature

*Clean up your git commit history before submitting the pull request!*

### CODE REVIEW

**Comment by PasiSa** [comment]
I decided to combine a couple of other related small fixes on the same pull request on the same go.

File: edit_course/course_forms.py [hunk 1]
```diff
@@ -88,6 +89,9 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
             instance=self.instance.active_model_solution_reveal_rule,
             prefix='model_solution',
         )
+        self.fields['model_answer'].queryset = CourseChapter.objects.filter(
+            course_module__course_instance=self.instance.course_instance,
+        )
```
Comment (Mankro) [inline edit_course/course_forms.py:94]: In addition, I was thinking that the model could validate that the model solution chapter belongs to the same course.

https://github.com/apluslms/a-plus/blob/3893d834479b0d8f4a301942ae98a894006b39eb/course/models.py#L1361-L1377

I can try that myself.

**Comment by Mankro** [comment]
@PasiSa During quick local testing, I encountered issues in the next/previous chapter links. In the first chapter of a module, the previous link is completely missing (it does not render even the name without a link: it is completely missing). In some chapters, the next/previous links only render the text without a link even though the student has access to the chapters/modules.

File: edit_course/course_forms.py (hunk 1 above)
Comment (Mankro) [reply edit_course/course_forms.py:94]: Done here: https://github.com/apluslms/a-plus/commit/a180fbf1cff90743776fadbfd6d51ba15823f046

**Pull request merged**